src/Servers/Kestrel/test/FunctionalTests/UnixDomainSocketsTests.cs

```diff
@@ -114,6 +116,73 @@ async Task EchoServer(ConnectionContext connection)
             }
         }
 
+#if LIBUV
+        [OSSkipCondition(OperatingSystems.Windows, SkipReason = "Libuv does not support unix domain sockets on Windows.")]
+#else
+        [MinimumOSVersion(OperatingSystems.Windows, WindowsVersions.Win10_RS4)]
+#endif
+        [ConditionalFact]
+        [CollectDump]
+        public async Task TestUnixDomainSocketWithUrl()
+        {
+            var path = Path.GetTempFileName();
+            var url = $"http://unix:/{path}";
+
+            Delete(path);
+
+            try
+            {
+                var hostBuilder = TransportSelector.GetWebHostBuilder()
+                    .UseUrls(url)
+                    .UseKestrel()
+                    .ConfigureServices(AddTestLogging)
+                    .Configure(app =>
+                    {
+                        app.Run(async context =>
+                        {
+                            await context.Response.WriteAsync("Hello World");
+                        });
+                    });
+
+                using (var host = hostBuilder.Build())
+                {
+                    await host.StartAsync().DefaultTimeout();
+
+                    // https://github.com/dotnet/corefx/issues/5999
+                    // .NET Core HttpClient does not support unix sockets, it's difficult to parse raw response data. below is a little hacky way.
+                    using (var socket = new Socket(AddressFamily.Unix, SocketType.Stream, ProtocolType.Unspecified))
+                    {
+                        await socket.ConnectAsync(new UnixDomainSocketEndPoint(path)).DefaultTimeout();
+   
+                        var httpRequest = Encoding.ASCII.GetBytes("GET / HTTP/1.1\r\nHost:\r\n\r\n");
```

Comment:
The following change should get you the 0-length read you're looking for.

```suggestion
                        var httpRequest = Encoding.ASCII.GetBytes("GET / HTTP/1.1\r\nHost:\r\nConnection: close\r\n\r\n");
```

Reply:
Thank you I really appreciate your hel!. it's working like a charm.